lib/tessel/wifi.js

```diff
@@ -163,7 +163,18 @@ Tessel.prototype.setWiFiState = function(enable) {
   return new Promise((resolve, reject) => {
     return this.simpleExec(commands.turnOnWifi(enable))
       .then(() => this.simpleExec(commands.commitWirelessCredentials()))
-      .then(() => this.simpleExec(commands.reconnectWifi()))
+      .then(() => {
+        return this.simpleExec(commands.reconnectWifi())
+          .then((result) => {
+            // check if the actual wifi radio (wireless.radio0) is disabled
+            if (result.includes("'radio0' is disabled")) {
+              return this.simpleExec(commands.turnRadioOn())
+                .then(() => this.simpleExec(commands.commitWirelessCredentials()))
+                .then(() => this.simpleExec(commands.reconnectWifi()))
+                .catch(reject);
+            }
```

Comment:
What happens if there is an error that doesnt include `"'radio0' is disabled"`? I think this needs to return `Promise.resolve()`?


(Sorry, I reposted this because I'm not sure if we're officially using Github Review yet)